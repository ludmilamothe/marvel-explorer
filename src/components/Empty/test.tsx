import { screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import Empty from '.';

const datas = {
  imagePath: '/images/no-search.svg',
  imageAlt:
    'A magnifying glass with sad face and with some bubbles in the background',
  children: 'Some text',
};

describe('<Empty />', () => {
  it('should render the empty correct', () => {
    renderWithProviders(<Empty {...datas} />);

    const image = screen.getByRole('img', { name: datas.imageAlt });
    const children = screen.getByText(datas.children);

    expect(children).toBeInTheDocument();

    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', datas.imagePath);
  });

  it('should render the empty styles correct', () => {
    const { container } = renderWithProviders(<Empty {...datas} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
