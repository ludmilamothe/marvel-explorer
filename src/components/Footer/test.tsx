import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/test/helpers';

import Footer from '.';

describe('<Footer />', () => {
  it('should render the footer content correct', () => {
    renderWithTheme(<Footer />);

    const linkText = 'Mauro de Souza';
    const copy = 'Data provided by Marvel. Â© 2020 MARVEL';

    const linkElement = screen.getByRole('link', { name: linkText });

    expect(linkElement).toHaveAttribute(
      'href',
      'https://www.linkedin.com/in/maurodesouzaa/'
    );

    expect(linkElement).toHaveAttribute('target', '_blank');

    expect(screen.getByText(copy)).toBeInTheDocument();
  });

  it('should render the footer styles correct', () => {
    const { container } = renderWithTheme(<Footer />);

    const footer = container.firstChild;

    expect(footer).toMatchSnapshot();
  });
});
