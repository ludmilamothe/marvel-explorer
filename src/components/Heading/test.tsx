import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/test/helpers';

import theme from 'styles/theme/default';
import Heading from '.';

describe('<Heading />', () => {
  it('should render the heading with children', () => {
    const text = 'Just a text to test';

    renderWithTheme(<Heading>{text}</Heading>);

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toBeInTheDocument();

    expect(heading).toMatchInlineSnapshot(`
      .c0 {
        color: #FFF;
        font-weight: 900;
        font-size: 2.4rem;
        max-width: 55rem;
        text-align: center;
      }

      @media (min-width:768px) {
        .c0 {
          font-size: 4.0rem;
        }
      }

      <h1
        class="c0"
      >
        Just a text to test
      </h1>
    `);
  });

  it('should render the heading with correct styles', () => {
    const text = 'Just a text to test';

    renderWithTheme(<Heading>{text}</Heading>);

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toHaveStyle({
      fontSize: theme.font.sizes.large,
      fontWeight: theme.font.xbold,
      color: theme.colors.white,
    });

    expect(heading).toHaveStyleRule('font-size', theme.font.sizes.xlarge, {
      media: '(min-width: 768px)',
    });
  });
});
