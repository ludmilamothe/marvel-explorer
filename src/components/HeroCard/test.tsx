/* eslint-disable @typescript-eslint/no-var-requires */

import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import { storageKey } from 'hooks/useFavoriteHeroes';
import HeroCard from '.';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter');
const push = jest.fn();

useRouter.mockImplementation(() => ({
  push,
}));

const heroData = {
  heroId: 123,
  heroName: 'Iron man',
  heroThumbnail:
    'http://i.annihil.us/u/prod/marvel/i/mg/9/c0/527bb7b37ff55.jpg',
};

describe('<HeroCard />', () => {
  beforeEach(() => {
    window.localStorage.clear();
  });

  it('should render the hero card correct', () => {
    renderWithProviders(<HeroCard {...heroData} />);

    expect(
      screen.getByRole('img', { name: heroData.heroName })
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', { name: heroData.heroName })
    ).toBeInTheDocument();

    expect(screen.getByLabelText(/add hero to favorites/i)).toBeInTheDocument();
  });

  it('should render the hero card styles correct', () => {
    const { container } = renderWithProviders(<HeroCard {...heroData} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add hero to favorites', () => {
    renderWithProviders(<HeroCard {...heroData} />);

    const addToFavoriteElement = screen.getByLabelText(
      /add hero to favorites/i
    );

    expect(addToFavoriteElement).toBeInTheDocument();
    fireEvent.click(addToFavoriteElement);

    const removeFromFavoriteElement = screen.getByLabelText(
      /remove hero to favorites/i
    );

    expect(removeFromFavoriteElement).toBeInTheDocument();
    expect(addToFavoriteElement).not.toBeInTheDocument();

    const storagedHero = JSON.parse(window.localStorage.getItem(storageKey)!);

    expect(storagedHero).toEqual([heroData]);
  });

  it('should remove hero to favorites', () => {
    window.localStorage.setItem(storageKey, JSON.stringify([heroData]));

    renderWithProviders(<HeroCard {...heroData} />);

    const removeFromFavoriteElement = screen.getByLabelText(
      /remove hero to favorites/i
    );

    expect(removeFromFavoriteElement).toBeInTheDocument();
    fireEvent.click(removeFromFavoriteElement);

    const addToFavoriteElement = screen.getByLabelText(
      /add hero to favorites/i
    );

    expect(removeFromFavoriteElement).not.toBeInTheDocument();
    expect(addToFavoriteElement).toBeInTheDocument();

    const storagedHero = JSON.parse(window.localStorage.getItem(storageKey)!);
    expect(storagedHero).toEqual([]);
  });

  it('should navigate to hero page', () => {
    const { container } = renderWithProviders(<HeroCard {...heroData} />);

    const heroCard = container.firstChild!;
    fireEvent.click(heroCard);

    expect(push).toBeCalledTimes(1);
    expect(push).toBeCalledWith(`/characters/${heroData.heroId}`);
  });
});
