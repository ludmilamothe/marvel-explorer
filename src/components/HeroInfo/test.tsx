import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import HeroInfo from '.';

const props = {
  id: 123,
  name: 'iron man',
  description:
    'Inventor Tony Stark applies his genius for high-tech solutions to problems as Iron Man, the armored Avenger.',
  wikiLink: 'https://google.com',
  comics: 123,
  series: 1234,
  stories: 12345,
};

describe('<HeroInfo />', () => {
  it('should render the HeroInfo correct', () => {
    renderWithProviders(<HeroInfo {...props} />);

    const link = screen.getByRole('link', { name: /see more/i });
    expect(screen.getByText(props.name)).toBeInTheDocument();

    expect(screen.getByText(props.description)).toBeInTheDocument();
    expect(screen.getByText(props.comics)).toBeInTheDocument();

    expect(screen.getByText(props.series)).toBeInTheDocument();
    expect(screen.getByText(props.stories)).toBeInTheDocument();

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', props.wikiLink);
  });

  it('should alter icon between add/remove favorite', () => {
    renderWithProviders(<HeroInfo {...props} />);

    let addToFavoriteElement = screen.getByLabelText(/add hero to favorites/i);

    expect(addToFavoriteElement).toBeInTheDocument();
    fireEvent.click(addToFavoriteElement);

    const removeFromFavoriteElement = screen.getByLabelText(
      /remove hero to favorites/i
    );

    expect(removeFromFavoriteElement).toBeInTheDocument();
    expect(addToFavoriteElement).not.toBeInTheDocument();

    fireEvent.click(removeFromFavoriteElement);

    addToFavoriteElement = screen.getByLabelText(/add hero to favorites/i);

    expect(removeFromFavoriteElement).not.toBeInTheDocument();
    expect(addToFavoriteElement).toBeInTheDocument();
  });
});
