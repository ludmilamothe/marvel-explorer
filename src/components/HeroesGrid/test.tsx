import { renderWithProviders } from 'utils/test/helpers';

import HeroesGrid from '.';

describe('<HeroesGrid />', () => {
  it('should render the HeroesGrid styles correct', () => {
    const { container } = renderWithProviders(<HeroesGrid />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
