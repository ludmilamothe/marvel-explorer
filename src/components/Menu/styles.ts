import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import * as FooterStyles from 'components/Footer/styles';

export const Container = styled.menu`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const MenuNav = styled.nav`
  display: none;
  align-items: center;
  flex-direction: column;
  justify-content: center;

  ${media.greaterThan('medium')`
    display: flex;
    flex-direction: row;
    justify-content: initial;
  `}
`;

export const MenuLink = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.xlarge};

    & + & {
      margin-top: ${theme.spacings.small};
    }

    ${media.greaterThan('medium')`
      position: relative;
      font-size: ${theme.font.sizes.large};

      & + & {
        margin-top: 0;
        margin-left: calc(${theme.spacings.large} * 2);
      }

      &:hover {
        &::after {
          content: '';
          position: absolute;
          display: block;
          background-color: ${theme.colors.secondary};
          height: 2px;
          animation: hoverAnimation 0.2s forwards;
        }

        @keyframes hoverAnimation {
          from {
            width: 0%;
            left: 50%;
          }
          to {
            left: 0%;
            width: 100%;
          }
        }
      }
    `}
  `}
`;

export const IconWrapper = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    width: 2.4rem;
    height: 2.4rem;
    cursor: pointer;

    ${media.greaterThan('medium')`
      display: none;
    `}
  `}
`;

type MenuFullProps = {
  isOpen: boolean;
};

export const MenuFull = styled.div<MenuFullProps>`
  ${({ theme, isOpen }) => css`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    height: 100vh;
    width: 100%;
    background-color: ${theme.colors.black};
    transition: opacity 0.2s;
    opacity: ${isOpen ? 1 : 0};
    pointer-events: ${isOpen ? 'all' : 'none'};
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    overflow: hidden;
    padding: 0 ${theme.spacings.small} 0;
    z-index: 20;

    ${IconWrapper} {
      position: absolute;
      top: ${theme.spacings.small};
      right: ${theme.spacings.small};
    }

    ${MenuNav} {
      display: flex;
      flex: 1;
      transition: transform 0.2s;
      transform: ${isOpen ? 'translateY(0)' : 'translateY(3.2rem)'};
    }

    ${FooterStyles.Container} {
      transition: transform 0.2s;
      transform: ${isOpen ? 'translateY(0)' : 'translateY(3.2rem)'};
    }

    ${media.greaterThan('medium')`
      display: none;
    `}
  `}
`;
