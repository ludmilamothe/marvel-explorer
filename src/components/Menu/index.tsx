import { useState } from 'react';
import Link from 'next/link';

import { Menu as MenuIcon } from '@styled-icons/material/Menu';
import { Close as CloseIcon } from '@styled-icons/material/Close';

import Logo from 'components/Logo';
import * as S from './styles';
import Footer from 'components/Footer';

const Menu = () => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <S.Container>
      <Logo />

      <S.MenuNav>
        <Link href="/" passHref>
          <S.MenuLink>Search</S.MenuLink>
        </Link>

        <Link href="/favorites" passHref>
          <S.MenuLink>Favorites</S.MenuLink>
        </Link>
      </S.MenuNav>

      <S.IconWrapper onClick={() => setIsOpen(true)}>
        <MenuIcon aria-label="Open menu" />
      </S.IconWrapper>

      <S.MenuFull aria-hidden={!isOpen} isOpen={isOpen}>
        <S.IconWrapper onClick={() => setIsOpen(false)}>
          <CloseIcon aria-label="Close menu" />
        </S.IconWrapper>

        <S.MenuNav>
          <Link href="/" passHref>
            <S.MenuLink>Search</S.MenuLink>
          </Link>

          <Link href="/favorites" passHref>
            <S.MenuLink>Favorites</S.MenuLink>
          </Link>
        </S.MenuNav>

        <Footer />
      </S.MenuFull>
    </S.Container>
  );
};

export default Menu;
