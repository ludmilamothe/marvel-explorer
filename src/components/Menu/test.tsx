import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from 'utils/test/helpers';

import Menu from '.';

describe('<Menu />', () => {
  it('should render the menu correct', () => {
    renderWithProviders(<Menu />);

    expect(screen.getByRole('img', { name: /marvel/i })).toBeInTheDocument();

    expect(screen.getByLabelText(/open menu/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/close menu/i)).toBeInTheDocument();

    const searchLinks = screen.getAllByText(/search/i);
    const favoritesLinks = screen.getAllByText(/favorites/i);

    expect(searchLinks).toHaveLength(2);
    expect(favoritesLinks).toHaveLength(2);

    searchLinks.forEach(link => {
      expect(link).toHaveAttribute('href', '/');
    });

    favoritesLinks.forEach(link => {
      expect(link).toHaveAttribute('href', '/favorites');
    });
  });

  it('should open/close full menu', () => {
    renderWithProviders(<Menu />);

    const openMenuElement = screen.getByLabelText(/open menu/i);
    const closeMenuElement = screen.getByLabelText(/close menu/i);

    const fullMenuElement = closeMenuElement.parentElement!.parentElement!;

    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('true');
    expect(fullMenuElement).toHaveStyle({ opacity: 0 });

    fireEvent.click(openMenuElement);

    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('false');
    expect(fullMenuElement).toHaveStyle({ opacity: 1 });

    fireEvent.click(closeMenuElement);

    expect(fullMenuElement.getAttribute('aria-hidden')).toBe('true');
    expect(fullMenuElement).toHaveStyle({ opacity: 0 });
  });

  it('should render the menu styles correct', () => {
    const { container } = renderWithProviders(<Menu />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
