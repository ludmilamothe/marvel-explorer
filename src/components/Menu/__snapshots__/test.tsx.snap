// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render the menu styles correct 1`] = `
.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  width: 100%;
  height: 3.2rem;
  max-width: 8rem;
}

.c11 {
  text-align: center;
  max-width: 57rem;
  border-top: 4px solid #C6A972;
  padding-top: calc(4.0rem / 2);
  color: #FFF;
  font-size: 1.2rem;
}

.c12 {
  margin-bottom: 1.2rem;
}

.c12 a {
  color: #FFF;
}

.c12 a:hover {
  color: #C6A972;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c13 {
  font-weight: 700;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  color: #FFF;
  font-weight: 700;
  font-size: 4.0rem;
}

.c4 + .c4 {
  margin-top: 2.4rem;
}

.c7 {
  color: #FFF;
  width: 2.4rem;
  height: 2.4rem;
  cursor: pointer;
}

.c9 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100vh;
  width: 100%;
  background-color: #202020;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  opacity: 0;
  pointer-events: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  padding-bottom: 2.4rem;
}

.c9 .c6 {
  position: absolute;
  top: 2.4rem;
  right: 2.4rem;
}

.c9 .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: translateY(3.2rem);
  -ms-transform: translateY(3.2rem);
  transform: translateY(3.2rem);
}

.c9 .c10 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: translateY(3.2rem);
  -ms-transform: translateY(3.2rem);
  transform: translateY(3.2rem);
}

@media (min-width:768px) {
  .c1 {
    height: 5rem;
    max-width: 12.5rem;
  }
}

@media (min-width:768px) {
  .c11 {
    font-size: 1.6rem;
  }
}

@media (min-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: initial;
    -webkit-justify-content: initial;
    -ms-flex-pack: initial;
    justify-content: initial;
  }
}

@media (min-width:768px) {
  .c5 {
    position: relative;
    font-size: 2.4rem;
  }

  .c4 + .c4 {
    margin-top: 0;
    margin-left: calc(4.0rem * 2);
  }

  .c5:hover::after {
    content: '';
    position: absolute;
    display: block;
    background-color: #C6A972;
    height: 2px;
    -webkit-animation: hoverAnimation 0.2s forwards;
    animation: hoverAnimation 0.2s forwards;
  }
}

@media (min-width:768px) {
  .c7 {
    display: none;
  }
}

<menu
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-label="Marvel"
      fill="none"
      height="100%"
      role="img"
      viewBox="0 0 125 50"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0.961538 0H0V0.961538H0.961538V0Z"
        fill="#EC1D24"
      />
      <path
        d="M0 0H125V50H0V0Z"
        fill="#EC1D24"
      />
      <path
        d="M30.2885 46.1539V3.84619H20.4722L16.9722 25.7068L13.5597 3.84619H3.84619V46.1539H11.5385V25.7616L14.9779 46.1539H19.0443L22.475 25.7616V46.1539H30.2885Z"
        fill="#FEFEFE"
      />
      <path
        d="M121.367 38.5183V46.1202H107.289V3.84619H114.87V38.5183H121.367ZM61.2097 24.576C60.6231 24.8587 60.0097 24.9991 59.4116 25.001V11.4048H59.45C60.0481 11.4 64.5116 11.5818 64.5116 18.1289C64.5116 21.551 62.9923 23.7097 61.2097 24.576ZM38.9904 32.9231L41.0895 14.8472L43.2673 32.9231H38.9904ZM105.966 11.5625V3.85292H84.4991L80.9654 29.602L77.475 3.85196H69.7327L70.5991 10.727C69.7068 8.96542 66.5375 3.85196 59.5606 3.85196C59.5154 3.85004 51.8106 3.85196 51.8106 3.85196L51.7808 41.3827L46.1385 3.85292L35.9991 3.84811L30.1606 42.7327L30.1625 3.85292H20.4597L16.9616 25.6962L13.5539 3.85196H3.84619V46.127H11.4904V25.75L14.9693 46.127H19.0327L22.4606 25.75V46.127H37.1981L38.0904 39.625H44.0231L44.9145 46.127L59.3827 46.1347H59.3923V46.127H59.4116V32.4058L61.1856 32.1462L64.8558 46.1366H72.3404L72.3385 46.127H72.3596L67.5414 29.7347C69.9818 27.927 72.7395 23.3462 72.0058 18.9625V18.9606C72.0135 19.0164 76.5529 46.1539 76.5529 46.1539L85.4529 46.1279L91.5366 7.81061V46.1279H105.966V38.5241H99.1164V28.8318H105.966V21.1106H99.1164V11.5616H105.966V11.5625Z"
        fill="#FEFEFE"
      />
    </svg>
  </div>
  <nav
    class="c2 c3"
  >
    <a
      class="c4 c5"
      href="/"
    >
      Search
    </a>
    <a
      class="c4 c5"
      href="/favorites"
    >
      Favorites
    </a>
  </nav>
  <div
    class="c6 c7"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c8"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    aria-hidden="true"
    class="c9"
  >
    <div
      class="c6 c7"
    >
      <svg
        aria-hidden="true"
        aria-label="Close menu"
        class="c8"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <nav
      class="c2 c3"
    >
      <a
        class="c4 c5"
        href="/"
      >
        Search
      </a>
      <a
        class="c4 c5"
        href="/favorites"
      >
        Favorites
      </a>
    </nav>
    <footer
      class="c10 c11"
    >
      <p
        class="c12"
      >
        Made with ❤ by
         
        <a
          href="https://www.linkedin.com/in/maurodesouzaa/"
          rel="noreferrer"
          target="_blank"
        >
          Mauro de Souza
        </a>
         
      </p>
      <p
        class="c13"
      >
        Data provided by Marvel. © 2020 MARVEL
      </p>
    </footer>
  </div>
</menu>
`;
